generate_directory_indexes.py

```diff
@@ -15,37 +22,68 @@
 
 #sys.setrecursionlimit(2100000000)
 
-parser = argparse.ArgumentParser()
-parser.add_argument("path", help="Path to index and write indexes to")
-parser.add_argument("--noop", help="Only print files to be created without writing them to disk", action="store_true")
-parser.add_argument('--verbose', '-v', help="Verbose output. Repeat (up to -vvv) for more verbosity", action='count')
-args = parser.parse_args()
-
-logger = logging.getLogger(__name__)
-if args.verbose == None:
-    logging.basicConfig(level=logging.ERROR)
-if args.verbose == 1:
-    logging.basicConfig(level=logging.WARNING)
-elif args.verbose == 2:
-    logging.basicConfig(level=logging.INFO)
-else:
-    logging.basicConfig(level=logging.DEBUG)
+script = os.path.basename(sys.argv[0])
+configuration = None
+
+def parse_arguments():
+    global configuration
+    global logger
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "path",
+        help="Top directory for writing index files.")
+    parser.add_argument(
+        "--file-metadata", "-f",
+        default=None,
+        help="A file containing data describing the tree to index.")
+    parser.add_argument(
+        "--metadata-delimiter", "-m",
+        default=";",
+        help="Character which seperates fields in the file metadata.")
+    parser.add_argument(
+        "--noop", "-n",
+        help="Only print files to be created without writing them to disk",
+        action="store_true")
+    parser.add_argument(
+        "--verbose", "-v",
+        help="Verbose output. Repeat (up to -vvv) for more verbosity",
+        action="count")
+    configuration = parser.parse_args()
+
+    logger = logging.getLogger(__name__)
+    logger_format = '%(filename)s: %(levelname)s %(message)s'
+
+    if configuration.verbose == 1:
+        logging.basicConfig(level=logging.WARNING, format=logger_format)
+    elif configuration.verbose == 2:
+        logging.basicConfig(level=logging.INFO, format=logger_format)
+    elif configuration.verbose > 2:
+        logging.basicConfig(level=logging.DEBUG, format=logger_format)
+    else:
+        logging.basicConfig(level=logging.ERROR, format=logger_format)
+
 
 def index_link(prefix, current_order_by, new_order_by, reverse_order):
     if current_order_by == new_order_by:
         return index_file_name(prefix, current_order_by, not reverse_order)
     else:
         return index_file_name(prefix, new_order_by, False)
 
-# use a templating library to turn a prefix and a list of contents into an HTML directory index
+
+####
```

Comment:
Same `"""` comment block issue as above. We should also move this comment into the function.
```py
def render_index(prefix, order_by, contents, reverse_order, base_path):
    """
    Use a templating library to turn a prefix and a list of contents
    into an HTML directory index
    """
```